GCC 4.9 compilation error in Geant4InputHandling.cpp

- OS version: SLC6
 - Compiler version: GCC 4.9.3 (CERN AFS)
    - source /afs/cern.ch/sw/lcg/contrib/gcc/4.9.3/x86_64-slc6/setup.sh
    - export PATH=/afs/cern.ch/sw/lcg/releases/LCG_87/Python/2.7.10/x86_64-slc6-gcc49-opt/bin:$PATH
    - export LD_LIBRARY_PATH=/afs/cern.ch/sw/lcg/releases/LCG_87/Python/2.7.10/x86_64-slc6-gcc49-opt/lib:$LD_LIBRARY_PATH
    - export CC=gcc
    - export CXX=g++

 - DD4hep version: https://github.com/AIDASoft/DD4hep/blob/master/DDG4/src/Geant4InputHandling.cpp#L434
 - Reproduced by: ILCSoft nightly build with gcc4.9.3
 - Error: 
      - /scratch/nbuilds/gcc49/2018-02-27/DD4hep/HEAD/DDG4/src/Geant4InputHandling.cpp: In lambda function:
      - /scratch/nbuilds/gcc49/2018-02-27/DD4hep/HEAD/DDG4/src/Geant4InputHandling.cpp:434:133: error: use of deleted function 'std::basic_stringstream<char>::basic_stringstream(const std::basic_stringstream<char>&)'
      - caller->debug("Rejecting PDGs: %s", [&rejectPDGs](){ std::stringstream str; for (int i: rejectPDGs) { str  << i << ", "; } return str;}().str().c_str());

 - Last update: https://github.com/AIDASoft/DD4hep/commit/0d8ae7116527b336222550031de05634a4982d3b#diff-9c36d2d64a01930018f1519715a3b4f9R434

--- a/DDG4/src/Geant4InputHandling.cpp
+++ b/DDG4/src/Geant4InputHandling.cpp
@@ -431,7 +431,7 @@ int dd4hep::sim::generatePrimaries(const Geant4Action* caller,
   auto const* primHandler = dynamic_cast<const Geant4PrimaryHandler*>(caller);
   auto const& rejectPDGs = primHandler ? primHandler->m_rejectPDGs : std::set<int>();
 
-  caller->debug("Rejecting PDGs: %s", [&rejectPDGs](){ std::stringstream str; for (int i: rejectPDGs) { str  << i << ", "; } return str;}().str().c_str());
+  caller->debug("Rejecting PDGs: %s", [&rejectPDGs]{ std::stringstream str; for (int i: rejectPDGs) { str << i << ", "; } return str.str();}().c_str());
 
   if ( interaction->locked )  {
     caller->abortRun("Locked interactions may not be used to generate primaries!",

